analyzer/queries/queries.go

```diff
@@ -166,9 +166,30 @@ var (
         SET
         height = excluded.height`
 
-	ConsensusBlockInsert = `
-    INSERT INTO chain.blocks (height, block_hash, time, num_txs, namespace, version, state_root, epoch, gas_limit, size_limit)
-      VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10)`
+	ConsensusBlockUpsert = `
+    INSERT INTO chain.blocks (height, block_hash, time, num_txs, namespace, version, state_root, epoch, gas_limit, size_limit, proposer_entity_id)
+      VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11)
+    ON CONFLICT (height) DO UPDATE
+    SET
+      block_hash = excluded.block_hash,
+      time = excluded.time,
+      num_txs = excluded.num_txs,
+      namespace = excluded.namespace,
+      version = excluded.version,
+      state_root = excluded.state_root,
+      epoch = excluded.epoch,
+      gas_limit = excluded.gas_limit,
+      size_limit = excluded.size_limit,
+      proposer_entity_id = excluded.proposer_entity_id`
+
+	ConsensusBlockSignersUpsert = `
+    INSERT INTO chain.blocks
+      (height, signer_entity_ids)
+    VALUES
+      ($1, $2)
+    ON CONFLICT (height) DO UPDATE
+    SET
+      signer_entity_ids = excluded.signer_entity_ids`
```

Comment:
Would this work if we used an `UPDATE` instead? We could then keep the `not null` constraints for the other `chain.blocks` columns. I _think_ they were originally removed to support updating the signers for the `from`-1 block (since the consensus analyzer didn't analyze the block before `from`, it won't have any of the fields populated). As a bonus, the migration becomes shorter/faster.